src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj

```diff
@@ -6,16 +6,28 @@
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <NoWarn>8002,NU1701</NoWarn>
     <Platforms>x64;x86</Platforms>
-    <PlatformTarget>AnyCPU</PlatformTarget>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendTargetFrameworkToOutputPath Condition="'$(BuildingInsideVisualStudio)' == 'true'">true</AppendTargetFrameworkToOutputPath>
     <OutputPath>$(BinDir)/crossgen2</OutputPath>
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
-    <RuntimeIdentifiers>linux-x64;linux-musl-x64;win-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>linux-x64;linux-musl-x64;win-x64;win-x86</RuntimeIdentifiers>
     <Configurations>Debug;Release;Checked</Configurations>
   </PropertyGroup>
 
+  <!-- Platform specific properties -->
+  <PropertyGroup Condition="'$(Platform)' == 'x64'">
+    <PlatformTarget>x64</PlatformTarget>
+    <TargetArchitecture>x64</TargetArchitecture>
+    <Prefer32Bit>false</Prefer32Bit>
+    <DefineConstants>TARGET_64BIT;TARGET_AMD64;$(DefineConstants)</DefineConstants>
```

Comment:
Are we giving up on a single compiler binary that can crosscompile for different architectures? We normally do `TargetDetails.Architecture` checks instead of compile-time `#if` checks within the compiler.

Reply:
At this point in time there's no such thing as x64-hosted JIT targeting x86. As I mentioned in yesterday CG2 planning meeting, it would be naturally the most elegant to be able to compile for all architectures from the same CG2 driver but that requires work from the JIT team to stand up the x64 -> x86 compilation mode; in several past discussions on the subject they pointed out this would be a non-trivial task in the order of man-months even though my hope is that the amount of work shouldn't be really enormous as they already support x64->arm32 (i.o.w. they have hopefully already managed to fix most architecture/bitness-specific conditional compilations in the C++ code).

Reply:
On the other hand, if we decide that we want to ship 32-bit Crossgen2 (presumably only targeting x86 / arm32) for whatever reason, it carries a set of problems of its own - most notably the previously unseen requirement to use both 32-bit and 64-bit dotnet SDK in the runtime repo and on the Helix machines. The main purpose of this PR is to identify these various obstacles and limitations to serve as a data point for our planning.

Reply:
My point is that we didn't have `#if TARGET_XXX` checks within the compiler in the past and that kept the compilers ability to multi-target (e.g. from x64 to ARM64 that RyuJIT supports). I don't see these constants being used anywhere - do you have plans to use them or can we stay on the `if (context.Target.Architecture == ...)` plan within the compiler?

Reply:
I hope that we won't need to use conditional compilation in the Crossgen2 compiler. The changes I made in the Crossgen2 msbuild project script were only intended to suppress producing its executable as "AnyCpu" because I need to make sure it's loaded into the 32-bit host, otherwise it cannot load the 32-bit JIT as described above. It may be the case that the project change is actually superfluous and / or incomplete as for now I switched Crossgen2 over to be launched via corerun. I'll revisit this once the change becomes fully functional and it starts to make sense discussing which of its parts are worth merging in.

Reply:
I've been looking at making the cross building jits just work. I've spent a couple days at this and it all looks pretty good. I agree with Michal though, we shouldn't be setting up this sort of define. There isn't really any value to it.